internal/service/advancedclustertpf/resource.go

```diff
@@ -99,80 +96,230 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
 	if diags.HasError() {
 		return
 	}
+	api := r.Client.AtlasV220240805.ClustersApi
 	patchReq := update.PatchPayloadTpf(ctx, diags, &state, &plan, NewAtlasReq)
+	projectID := plan.ProjectID.ValueString()
+	clusterName := plan.Name.ValueString()
+	var cluster *admin20240805.ClusterDescription20240805
+	var advConfig *admin20240805.ClusterDescriptionProcessArgs20240805
```

Comment:
similar comment as in api, if we using "legacy" for admin20240530, i would use something for admin20240805